test/end2end_test.go

```diff
@@ -5342,7 +5342,7 @@ func testRPCTimeout(t *testing.T, e env) {
 		ResponseSize: respSize,
 		Payload:      payload,
 	}
-	for i := -1; i <= 10; i++ {
+	for i := 1; i <= 10; i++ {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(i)*time.Millisecond)
```

Comment:
This is even worrysome.  If it takes >1ms before the timeout is computed by grpc, then it could fail.  It seems like there's probably something in our code that we should change that prevents the client from sending a negative deadline instead.

Reply:
Made the fix to validate the timeout on the client side and return a `DeadlineExceeded` error.